public/js/uploader.js

```diff
@@ -273,61 +292,91 @@ Uploader.upload = function(bins, callback) {
 // Apply a key mapping.
 // Converts arrays from .csv file input
 // into the corresponding JSON objects
-Uploader.formatData = function(data){
+// flag: whether or not multiple key mappings need to be found.
+Uploader.formatData = function(data, flag){
 
   UI.alert("Filtering Valid Data.");
 
   var acc = [];
+  if (flag > 1) {
+    _.each(data, function(current){
 
-  _.each(data, function(current){
+      var action = current[0];
 
-    var action = current[0];
+      // Get key mapping for the current entry
+      var keyMapping = getKeyMapping(settings.game, action);
 
-    // Get key mapping for the current entry
-    var keyMapping = getKeyMapping(settings.game, action);
+      // If we have a key mapping, assign keys to the current data
+      if (keyMapping){ 
+        var entry = assignKeys(current, keyMapping.columns) 
+      };  
 
-    // If we have a key mapping, assign keys to the current data
-    if (keyMapping){ 
-      var entry = assignKeys(current, keyMapping.columns) 
-    };  
-  
-   // Return data that was converted.
-   if (entry) { acc.push(entry); }
+      // Return data that was converted.
+      if (entry) { acc.push(entry); }
+
+    });
+  } else {
+    
+      var keyMapping = getKeyMapping(settings.game, settings.game);
+      _.each(data, function(current) {
+        
+        if (keyMapping) {
+          //console.log("calling assignKeys on " + current + " and " + keyMapping);
+          var entry = assignKeys(current, keyMapping.colums)
+        };
 
- });
+        if (entry) { acc.push(entry); };
 
-UI.alert(acc.length + " of " + data.length + " Entries Valid.")
+      });
+  }
 
-return acc;
+  UI.alert(acc.length + " of " + data.length + " Entries Valid.")
+  if (acc.length != data.length) {
+    UI.alert("For invalid entries, make sure each entry has "
+      + "X and Y position data, and a timestamp.")
+  }
+  return acc;
 }
 
+
+
 Uploader.fillMissingData = function(data) {
   var entries = [];
-
+  
   // Iterate through entries
+  var playerID = '';
+  if (data[0].playerID) {
+    playerID = data[0].playerID;
+  } else {
+    playerID = prompt("The data seems to be missing a player ID."
+      + " Please indicate one to use.");
+  }
   for (var i = data.length - 1; i >= 0; i--) {
-    
+
     // Current entry
     var current = data[i];
-
     // Ensure data has time, x, and y data
     // If not, get it from the last user entry that does
-    // Note - this is only to fix a data error with the client's current data
-    // set. It is not meant to be a full fledged feature.
     if (current.posX &&  current.posY && current.timestamp && current.area) {
 
       // 
       data[i] = current;
-    
+
     // Else, if data is missing:
     // Populate the missing data
+  } else {
+    if (settings.game == "Gaze") {
```

Comment:
Instead of hard-coding the "Gaze" string, lets use the function you wrote above:

```javascript
if (numberOfGameMappings(selectedGame) > 1) {
```